### PR TITLE
feat(client): unify GA and add to donation

### PR DESCRIPTION
**Pull request by ahmaxed**
This pr would add specific ga.event and ga.pageview/ga.modalview for all of the following donation types:

1. donation modal stripe
2. donation page
3. year end gift stripe and paypal donation
4. certification

### CODE REVIEW

**Comment by ahmadabdolsaheb** [comment]
The logic for the GA takes place in the parent of the forms in handleProcessing.

@raisedadead, if you prefer having GA outside the components, we could add a redux action with specific event payload and have a saga listen for it and call ga.event there. this would only require connecting 2-3 functional components to redux. please let me know if you would like to take that route.

Additionally, the donation related page views and modals are reported using ga.pageview and ga.modalview. let me know if you would like to have events for them.

**Comment by raisedadead** [comment]
Yes to all of your queries. Thanks for cleaning this up, looks much sane now.

**Comment by ahmadabdolsaheb** [comment]
@raisedadead this should be ready.

**Comment by raisedadead** [comment]
I thought you wanted to write redux actions for this.

File: client/src/redux/index.js [hunk 1]
```diff
@@ -95,6 +98,9 @@ export const appMount = createAction(types.appMount);
 export const tryToShowDonationModal = createAction(
   types.tryToShowDonationModal
 );
+
+export const reportGaEvent = createAction(types.reportGaEvent);
```
Comment (ahmadabdolsaheb) [inline client/src/redux/index.js]: @raisedadead Here is the action. just removed calling the react.ga events from the components and relocated them to a SAGA.

Comment (ahmadabdolsaheb) [reply client/src/redux/index.js]: please clarify if you had something else in mind.

Comment (raisedadead) [reply client/src/redux/index.js]: What would be these calls: https://github.com/freeCodeCamp/freeCodeCamp/pull/37984/files#diff-e5939f8d99bc04b87018776785947561R74

Comment (ahmadabdolsaheb) [reply client/src/redux/index.js]: @raisedadead, only ga.events were included in the action. 
Now I included all GA is called from there and should be ready for full review.

**Review by raisedadead — APPROVED** [review]

**Comment by raisedadead** [comment]
Now this is unified, please feel free to add anyplace you deem fit. Nice work.

**Pull request merged**